### PR TITLE
[UC03#15] Implemented RPC GetGameCard() that enables visitors to retrieve a specific card by its id

### LINKED ISSUE #15: Connessione RemoteService FetchCard nel caso l'assunzione sia errata [UC03]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  La documentazione è stata aggiornata
- [x]  Il Code Review è stato fatto
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #15 to enable visitors to retrieve detailed informations about a specific card by its unique identification and game type. This allows for the convenient bookmarking of individual card pages.
Supplied with coverage for unit testing.

### Additional work done on this task:
I have added unit tests for ListenerImpl and file loading on the Database. I have also refactored HomeViewImpl to organize the order of functions and remove unused function passed to `GameFiltersWidget` and added a id to CardPlace in order to include the id in the URI.

### CODE REVIEW

**Review by alessiacrimaldi — CHANGES_REQUESTED** [review]
Dear Alessio, thanks for your PR. I'd like to bring your attention on just two points, so please look at the comment that I left.
Moreover, I'd suggest you to enrich your `PR description` with a brief description of the code that you adjust which does not concern this specific task. In other words, let the PR be more understandable describing all the work you do, not just the work you're supposed to do pursuing the task.

File: src/main/java/com/aadm/cardexchange/shared/models/CardDecorator.java [hunk 1]
```diff
@@ -1,31 +1,39 @@
 package com.aadm.cardexchange.shared.models;
 
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class CardDecorator implements Card {
     private static final long serialVersionUID = -6914752354287411438L;
-    private CardImpl wrappee;
```
Comment (alessiacrimaldi) [inline src/main/java/com/aadm/cardexchange/shared/models/CardDecorator.java:7]: You should fix this typo in this file. I think it was supposed to be 'wrapped'. Keep up the attention to detail.

**Comment by alessioarcara** [comment]
I apologize, I realize now that I have not provided a detailed description of the added work. Thank you for pointing that out. ✌🏻

**Review by alessiacrimaldi — APPROVED** [review]
Ok, all right.

**Pull request merged**